### DISCUSSION TITLE
How to use the direction pad of the xbox360 controller？

### DISCUSSION THREAD

**Discussion by myhewen** [pybricks/support#1537] [Q&A]
It is not possible to set it up, down, left, and right

**Comment by laurensvalk** [comment]
Can you show the code or block example you've tried? Are the other buttons working?

**Comment by myhewen** [comment]
For example, I want to turn on the light with up control and turn off the light with down control, but there is only one button option, so the button that can realize the four functions of up, down, left and right can only realize one function.
![Uploading WechatIMG109.jpg…]()

**Comment by myhewen** [comment]
For example, I want to turn on the light with up control and turn off the light with down control, but there is only one button option, so the button that can realize the four functions of up, down, left and right can only realize one function.

**Comment by laurensvalk** [comment]
The direction pad gives a _number_ for the direction. So not just a true/false value as with normal buttons.

![image](https://github.com/pybricks/support/assets/12326241/0a294e52-f01a-41d1-bd90-7f981a18a58c)

So you can check `== 1` for up, `== 2 ` for up right, and so on.

**Comment by myhewen** [comment]
Thank you! How do you do that if you use code blocks? Can you give me a picture of the code block? Thank you!

**Reply by laurensvalk** [reply]
Sure! Here is an example that will change the hub light to blue for up, green for up/right, red for right, and violet for everything else.

You can add more cases or actions as you like:

![image](https://github.com/pybricks/support/assets/12326241/16a6d4b8-01a5-408a-bec1-0a4cf6141f2f)

**Comment by myhewen** [comment]
Thank you! One more question to ask:
How to use block code and xbox controller to achieve LEGO 42114 and 42131 gear shift operation? Can you give me a picture tutorial? Thank you!
